packages/svelte/src/internal/client/types.d.ts

```diff
@@ -46,15 +46,28 @@ export type ComponentContext = {
 	};
 };
 
-export type Signal<V = unknown> = {
+export type SourceSignal<V = unknown> = {
+	/** Signals that read from the current signal */
+	consumers: null | ComputationSignal[];
+	/** The associated component if this signal is an effect/computed */
+	context: null | ComponentContext;
+	/** For value equality */
+	equals: null | EqualsFunctions;
+	/** The types that the signal represent, as a bitwise value */
+	flags: SignalFlags;
+	/** The latest value for this signal, doubles as the teardown for effects */
+	value: V;
+};
+
+export type ComputationSignal<V = unknown> = {
```

Comment:
Is this an extension of the Signal interface? If so, should we make these interfaces so we can use `extends`?

Reply:
There is no `Signal` interface. `Signal` is a union of the two shapes.

Reply:
Sorry I meant `SourceSignal`, which looks like a strict subset of `ComputationSignal`